src/Analysis/Ast/Impl/Analyzer/Symbols/SymbolCollector.cs

```diff
@@ -229,6 +242,39 @@ internal sealed class SymbolCollector : PythonWalker {
                     DiagnosticSource.Analysis));
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="fd"></param>
+        /// <returns></returns>
+        private bool IsRedefinedByDecorator(FunctionDefinition fd) {
+            if (fd.Decorators != null) {
+                foreach (var d in fd.Decorators.Decorators) {
+                    switch (d) {
+                        case NameExpression n:
+                            if (fd.Name.Equals(n.Name)) {
```

Comment:
I think `fd.Name` can be null in `def (a, b, c)`

Reply:
I checked this, doesn't seem to be the case.

Reply:
I'll leave a check in there however.